tree/node.go

```diff
@@ -0,0 +1,101 @@
+package tree
+
+import (
+	"github.com/sasswart/gin-in-a-can/errors"
+)
+
+var _ NodeTraverser = &Node{}
+
+type Metadata map[string]string
+type Node struct {
+	basePath string
+	parent   NodeTraverser
+	metadata Metadata
+	Name     string
+}
+type Noder interface {
```

Comment:
We should document how a Noder is used in the rest of the codebase. What is the indended purpose of this intergace?